pkg/reporting/send.go

```diff
@@ -47,10 +53,27 @@ func (s eventSender[E]) senderLoop(ctx context.Context, logger *zap.Logger) {
 		case <-ctx.Done():
 			logger.Info("Received notification that events submission is done")
 			final = true
-		case <-ticker.C:
+			// finish up any in-progress batch, so that we can send it before we exit.
+			s.queue.finishOngoing()
+		case <-s.batchComplete.Wait():
+			s.batchComplete.Awake() // consume this notification
```

Comment:
question: why is that needed? 

sidenote, nitpick: comment is a great example of "what we do" statement while it would be more useful to have "why we do" statement

Reply:
It's needed because that's the interface for that notification type.

Although, thinking about it again, I think we can use something simpler and just read from a buffered channel.

Implemented that fix in 50923447360fda103dddd566472f5880963c7412.